src/components/AddBook.js

```diff
@@ -1,24 +1,57 @@
-import React from 'react';
-
-const AddBook = () => (
-  <form>
-    <h4>Add New Book</h4>
-    <input
-      type="text"
-      placeholder="Add Title ..."
-      name="title"
-      className="add-title"
-    />
-    <input
-      type="text"
-      placeholder="Add Author"
-      name="author"
-      className="add-author"
-    />
-    <button type="submit" className="add-submit">
-      ADD BOOK
-    </button>
-  </form>
-);
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { nanoid } from '@reduxjs/toolkit';
+import { addBook } from '../redux/books/booksSlice';
+
+const AddBook = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const changeTitle = (event) => {
+    setTitle(event.target.value);
+  };
+
+  const changeAuthor = (event) => {
+    setAuthor(event.target.value);
+  };
+
+  const submitForm = (event) => {
+    event.preventDefault();
+    const id = nanoid;
+
+    dispatch(addBook({
+      item_id: id, title, author, category: '',
+    }));
+
+    setTitle('');
+    setAuthor('');
+  };
+
+  return (
+    <form onSubmit={submitForm}>
+      <h4>Add New Book</h4>
+      <input
+        type="text"
+        placeholder="Add Title ..."
+        name="title"
+        className="add-title"
+        value={title}
+        onChange={changeTitle} // onChange is a prop that takes a function
+      />
+      <input
+        type="text"
+        placeholder="Add Author"
+        name="author"
+        className="add-author"
+        value={author}
+        onChange={changeAuthor}
+      />
+      <button type="submit" className="add-submit">
```

Comment:
- [ ] Please 🙏🏻 adhere to the project requirements, according to which you should add the `onClick` event on the **Button** component, which will help to add or remove the either new book or existing book into the `redux` store using the reducer. Right now you are using the `onClick` event but using `jsx` button tag, not the component, which is fine but not fulfilling the requirement. You should create a Button.js file for creating the component and then import that component and add the `click` event to make things work as it working right now.

### Requirement
![Screenshot from 2023-08-23 21-53-57](https://github.com/MohamedRoshdy2021/Book-store/assets/66476329/c7ad4113-ea5e-4e3f-9794-d95fa9e084dd)